packages/jsii-dotnet-runtime/src/Amazon.JSII.Runtime/Services/ReferenceMap.cs

```diff
@@ -46,8 +48,9 @@ ConstructorInfo GetByRefConstructor()
                 Type type = _types.GetClassType(byRefValue.FullyQualifiedName);
                 if (type == null)
```

Comment:
Just make this:

```
if (type == null || type.IsAbstract) {
    ...
}
```

And remove the check later in the method.

Reply:
Good catch. Will change.